backend/dataall/db/models/ShareObject.py

```diff
@@ -33,5 +33,7 @@ class ShareObject(Base):
     updated = Column(DateTime, onupdate=datetime.now)
     deleted = Column(DateTime)
     confirmed = Column(Boolean, default=False)
+    requestPurpose = Column(String, nullable=True)
+    rejectPurpose = Column(String, nullable=True)
```

Comment:
Although it is handled by `{share.rejectPurpose || '-'} ` in frontend, so it is not needed

Reply:
I am inclined to leave it as `null` rather than pass a default for these 2 columns

For `submit_share_object()` - the reason I include the line `share.rejectPurpose = ""` is for instances when a share is

1. Submitted by Requester
2. Rejected by Approver with Reason 1
3. Re-Submitted by Requester (changing Share Items requested)
4. Approver is able to reject again with new Reason 2

Maybe a better design would be to always allow the Approver to edit the reject reason and always allow the Requester to edit the request reason but I am not sure if that level of edit-ability is needed / would be used. What do you think?

Reply:
Added the edit ability in the latest commit where Requesters can edit requestPurpose and Approvers can edit rejectPurpose respectively